streams/src/test/java/org/apache/kafka/streams/integration/OptimizedKTableIntegrationTest.java

```diff
@@ -133,11 +133,6 @@ public void shouldApplyUpdatesToStandbyStore() throws Exception {
             kafkaStreams1WasFirstActive = false;
         }
 
-        // Assert that no restore has occurred, ensures that when we check later that the restore
-        // notification actually came from after the rebalance.
-        assertThat(listener.startOffset, is(equalTo(0L)));
-        assertThat(listener.totalNumRestored, is(equalTo(0L)));
```

Comment:
I am not sure if this is the right fix. Note that "restoring" and "standby task update" are two different things, and a standby task should. never use the "restore" code path what this assertion verifies.

What could explain a restore is the migration of the active task from one instance to the other. However, this should actually not happen either. Could you reproduce the issue locally? We recently worked on rebalancing so maybe the issue is with regard to that.

Reply:
Sorry @mjsax , I tried to trace the code to find out why that would happen, but I still can't figure it out. Do you have any suggestion for it? If not, I think we can have more clear message output when this assert failure happen again. I'm not sure if the info is enough if the error happened again. And not sure if you have any thoughts about it? 

It'll look like this when assert failure.
![image](https://user-images.githubusercontent.com/43372967/82539162-8dc6f680-9b7f-11ea-979a-24707df09245.png)

Thanks.

Reply:
I am not 100% sure about the root cause of the issue... If you have a good suggestion for a better error message, that would be great. I have not idea how we could improve it atm.